### PR TITLE
[data grid] Fix typo on the `rowSelectionPropagation` feature

### PR DESCRIPTION
**Pull request by flaviendelangle**
Noticed those while working on #14899

## Other proposed change

I'm also not a fan of having that many flat params on `findRowsToSelect` and `findRowsToDeselect`.
I feel like it would be a lot more readable:

1. Passing `props.rowSelectionPropagation` to the function  (right now if you are searching the codebase, this concept can be named `rowSelectionPropagation.descendants` or `autoSelectDescendants` depending on where you are looking for)
2. Moving to an object (it's subjective, but for this method I think the perf overhead is negligible, feel free to ignore this one)

```tsx
findRowsToSelect({
    apiRef,
    tree,
    id,
    rowSelectionPropagation,
    addRow,
});
```

### CODE REVIEW

File: packages/x-data-grid/src/models/props/DataGridProps.ts [hunk 1]
```diff
@@ -845,9 +845,9 @@ export interface DataGridProSharedPropsWithDefaultValue {
    * - Selecting a parent will auto-select all its filtered descendants.
```
Comment (MBilalShafi) [inline packages/x-data-grid/src/models/props/DataGridProps.ts:845]:
```suggestion
   * - Selecting a parent auto-selects all its filtered descendants.
```

File: packages/x-data-grid/src/models/props/DataGridProps.ts [hunk 2]
```diff
@@ -845,9 +845,9 @@ export interface DataGridProSharedPropsWithDefaultValue {
...
    * - Deselecting a parent will auto-deselect all its filtered descendants.
```
Comment (MBilalShafi) [inline packages/x-data-grid/src/models/props/DataGridProps.ts:846]:
```suggestion
   * - Deselecting a parent auto-deselects all its filtered descendants.
```

**Review by MBilalShafi — APPROVED** [review]
Thank you 💙 

Nitpick: Suggested an alternate grammatical sense since I've seen the usage of 'will' to be discouraged. Feel free to drop the suggestions.

Regarding the other suggestions, totally agree on the first one, will handle in a follow-up PR.
On the second one, we lately started to follow more and more the pattern of passing arguments rather than an options object. But I'm fine with moving to the object pattern in this specific case if there's a clear win.

File: docs/data/data-grid/row-grouping/row-grouping.md [hunk 3]
```diff
@@ -326,8 +326,8 @@ When `rowSelectionPropagation.descendants` is set to `true`.
 
 When `rowSelectionPropagation.parents` is set to `true`.
 
-- Selecting all the filtered descendants of a parent would auto-select the parent.
-- Deselecting a descendant of a selected parent would auto-deselect the parent.
+- Selecting all the filtered descendants of a parent will auto-select the parent.
+- Deselecting a descendant of a selected parent will auto-deselect the parent.
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/data-grid/row-grouping/row-grouping.md:330]: Not sure `auto-deselects` is a very valid verb :angel: 
How about?

```suggestion
- Deselecting a descendant of a selected parent automatically deselect the parent.
```

[the remark this answers is not in the document]
Comment (MBilalShafi) [reply docs/data/data-grid/row-grouping/row-grouping.md:330]: However I'm not sure which one would be better:
`Deselecting a descendant of a selected parent automatically deselects the parent.`
vs
`Deselecting a descendant of a selected parent deselects the parent automatically.`

To me, the second one looks like a more natural tone.
What do you think?

[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/data-grid/row-grouping/row-grouping.md:330]: I agree that the 2nd one is more natural
I'll update all the lines :+1:

[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/data-grid/row-grouping/row-grouping.md]: Done :+1:

**Pull request merged**